### PR TITLE
Default: Add nodebox mese post light usable as a growlamp 

### PR DESCRIPTION
**Pull request by paramat**
Light level 14 to be able to grow plants.
3 returned from crafting with 1 glass node, 3 mese crystals and 1
wood plank.
Seamlessly placable on appletree wood fence posts.
////////////////////////////////////////////////

screenshot_20170131_154424

screenshot_20170131_154608

^ Here also placed on top of fence post

screenshot from 2017-01-01 17 10 59

screenshot_20170127_190342

^ Alternative texture

Addresses #1478 

To have a more natural look as requested and not be too modern this has a wooden supporting post, however to also be suitable for modern builds this has a simple square post shape, which also echos the shape of the new 3D torches.
Since it's mostly wood and has a wood top i have given it wood plank groups and wood sounds.
Seamlessly placable on appletree fence posts to create fence lights or post lights of any height.

Because light sources do not have face shading applied i have darkened the wood side textures to match to the sides of fence posts and normal wood planks.
Recipe has 'glass over mese' to be intuitive and similar to the new meselamp recipe of #1481 
It is 1 node tall so any falling nodes land on it properly instead of hovering as entities.
The recipe results in 1 light per mese crystal, consistent with #1481

### CODE REVIEW

**Comment by tenplus1** [comment]
This should be a mod and not in default game.

**Comment by C1ffisme** [comment]
@tenplus1 
> Addresses #1478 

I think your missing some context.

**Comment by paramat** [comment]
VanessaE, be patient!

**Comment by paramat** [comment]
Moved the meselamp recipe change to a new PR, it should be a separate commit and needs to be merged quickly #1481

**Comment by stujones11** [comment]
If you want to make the recipe less expensive then why not jusu use mese fragments instead of the crystal. A mese torch could be simply made with 1 stick + 1 mese fragment to be consistent with the default torch.

**Comment by paramat** [comment]
Because the one glass node should return multiple lights, as that's a lot of glass and this node only has a samll amout of glass.
A mese crystal fragment is 1/9th of a crystal so is too cheap.
This way speeds up crafting of multiple lights.

**Comment by stujones11** [comment]
Why do we need to use glass? For the mese lamp, sure but not for a simple torch.

**Comment by paramat** [comment]
Because it's a mese light just like the meselamp, a mese crystal is raw material and does not glow (see the mese ore node).

**Comment by stujones11** [comment]
lol, I think you are just splitting hairs now, it was only a suggestion :)

**Comment by paramat** [comment]
That's cool.

**Comment by SmallJoker** [comment]
The side texture of the torch looks quite redundant to me. I would prefer to use the transparent inventory texure on both places, as you already implied in the main post.

**Comment by Napiophelios** [comment]
I would like to suggest you adapt Calinou's Carbone Torches for this
https://forum.minetest.net/viewtopic.php?t=10465
https://github.com/calinou/carbone_torches/

That way players can have ceiling and wall mounted mese lanterns as well.
screenshot_20170101133644

**Comment by stujones11** [comment]
Another likely controversial idea would be to make torches more api like so torches can be registered with different light-levels and/or textures. The mese 'torch' could then be placed exactly as a normal torch which may be helpful for those with underground farms.

**Comment by sofar** [comment]
How about, instead of adding all sorts of new recipes for weird nodes that we can't agree on, we instead do a change to farming where we (1) allow growing of plants and saplings for lower light leves and (2) increase the time needed to grow at lower light leves?

Since everything is now nodetimer based, it would be trivial to adjust the timers to add extra time when the light level is below 14. We could easily stretch it to still allow growth at light level 11 or 10, but just make it take 4-5x longer, or some value that's reasonable.

**Comment by C1ffisme** [comment]
@sofar I think that trees should still require at least a torch to grow, but that sounds like a good balance.

**Comment by paramat** [comment]
Yeah i'll use the 1 texture for both uses.

> instead of adding all sorts of new recipes for weird nodes that we can't agree on

That's a rather negative view of this PR, it's not controversial and nicer-looking mese lights have been requested by a few people recently during the lighting discussions.

>  we (1) allow growing of plants and saplings for lower light leves and (2) increase the time needed to grow at lower light leves?

I oppose this, because torches have never been bright enough to grow plants throughout the history of Minetest, which makes sense as they do not have the full spectrum light to be growlamps whatever brightness they are.
Plants require something special, and mese is ideal as that special something.
Perhaps you would agree that growing plants in dark areas requires a little more effort than a cheap torch.

>  Why was 13 chosen anyway?

Not sure, maybe celeron55 knows, it may or may not be coincidence that this is the level that prevents growing by torches. I can imagine celeron55 may not have been thinking about underground farming and just wanted to make sure that players needed daylight to grow stuff.

**Comment by stujones11** [comment]
FWIW I am happy with this as a solution and was merely suggesting an alternative, even less expensive recipe but it's not something I feel particularly strongly about.

Growing by default torches, however, would greatly change the game mechanics and may well have other unforeseen consequences. like stuff growing where it shouldn't.

File: mods/default/crafting.lua [hunk 1]
```diff
@@ -640,6 +640,15 @@ minetest.register_craft({
 })
 
 minetest.register_craft({
+	output = 'default:mese_light 3',
```
Comment (rubenwardy) [inline mods/default/crafting.lua]: Isn't the code style to use `"`? Even though the surrounding code doesn't

File: mods/default/nodes.lua [hunk 2]
```diff
@@ -2078,6 +2079,27 @@ minetest.register_node("default:meselamp", {
 	light_source = default.LIGHT_MAX,
 })
 
+minetest.register_node("default:mese_light", {
+	description = "Mese Light",
+	tiles = {"default_mese_light_top.png", "default_mese_light_top.png",
+		"default_mese_light_side.png"},
+	inventory_image = "default_mese_light_inv.png",
+	wield_image = "default_mese_light_inv.png",
```
Comment (rubenwardy) [inline mods/default/nodes.lua]: 90% sure inventory_image is used if there's no wield_image

File: mods/default/crafting.lua (hunk 1 above)
Comment (paramat) [inline mods/default/crafting.lua]: Yeah i guess we should be changing to "", we are elsewhere.

File: mods/default/nodes.lua (hunk 2 above)
Comment (paramat) [inline mods/default/nodes.lua]: Ok.

**Comment by paramat** [comment]
Updated, also using 1 texture for both 'side' and inventory.

**Comment by sofar** [comment]
Is it possible to make the "stick" part 2px instead of 4? I find the appearance quite 'raw'.

**Comment by paramat** [comment]
I find a narrow pole ugly and it doesn't have enough surface area to have a nice amount of wood texture.
Bollard lights look much better being the same width from top to bottom. But i can add a little more detail, i'll try.
I'm going for this look:

a-few-basics-on-bollard-light-installation

**Comment by Zeno-** [comment]
:-1: It's way to bright and doesn't look natural

It's also not consistent with the "medieval" goal of MTG

**Comment by C1ffisme** [comment]
> It's also not consistent with the "medieval" goal of MTG

As I've said before, MTG does not necessarily have any kind of time period, since we don't have any features at this point to suggest one.

**Comment by paramat** [comment]
It's the same texture brightness as the meselamp.

> It's also not consistent with the "medieval" goal of MTG

MTG is not restricted to medieval nodes, it has modern nodes too.
I was first thinking a steel pillar instead but deliberately chose wood specifically because you and others were asking for something with a natural feel.

**Comment by rubenwardy** [comment]
I don't see this pr as controversial, it's the saplings light level that is controversial

**Comment by Zeno-** [comment]
The style of this lamp is completely "wrong" for minetest_game. If it was in homedecor it'd be fine.

**Comment by paramat** [comment]
I'm going to work on the texture, add more detail. Light sources have no node face shading so i will darken the wood areas to match other wood nodes.
I agree the texture is not quite there yet.

**Comment by JurajVajda** [comment]
I see like this would want to have all kinds of wood types/colors (like fences) :)
screenshot_20170104_170056

**Comment by rubenwardy** [comment]
Does this fit on top of a fence post

**Comment by JurajVajda** [comment]
@rubenwardy yes, maybe even torch like drawtype would be nice

**Comment by paramat** [comment]
Updated to a much better texture with glass faceting and darkened wood texture to compensate for the lack of rendered face shading on light source nodes. New screenshots in first post.
Also now rotatable by screwdriver.

Somethng to consider though: should this have a more descriptive node name such as 'mese bollard light'? To allow more mese lights to be added in future?

**Comment by paramat** [comment]
screenshot_20170106_013711

Nice, i didn't realise these sat on top of a fence post. The problem now is that with my darkened texture there is a texture mismatch between fence post and light, so perhaps better to use an unaltered wood texture.

I feel this may need a different name, not a generic name, how about 'mese_post_light'?

**Comment by JurajVajda** [comment]
@paramat i have used slightly modified fence texture in my screenshot so it fits perfectly, the only thing is that it is asking for more wood colors/types - depends from what wood you are crafting it - so it can fit with all wooden fence types and colors.

**Comment by paramat** [comment]
Yes i was wrong, the fence posts are face shaded so the texture is close, but i have just made it even closer, with a second darker texture for EW faces.

**Comment by paramat** [comment]
Oops just need to remove facedir for the faces to orient correclty.

**Comment by paramat** [comment]
Done.

**Comment by JurajVajda** [comment]
Looks good :+1:

**Comment by paramat** [comment]
Changed name to default:mese_post_light.

**Comment by sofar** [comment]
Torches only come in 1 wood variety, so I don't think we should add more wood type variants for this node.

**Comment by paramat** [comment]
I'll tweak the texture again.

File: mods/default/nodes.lua [hunk 3]
```diff
@@ -2078,6 +2079,27 @@ minetest.register_node("default:meselamp", {
...
+		type = "fixed",
+		fixed = {
+			{-2 / 16, -8 / 16, -2 / 16, 2 / 16, 8 / 16, 2 / 16},
+		},
+	},
+	paramtype = "light",
+	light_source = default.LIGHT_MAX,
+	sunlight_propagates = true,
+	is_ground_content = false,
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2},
```
Comment (DS-Minetest) [inline mods/default/nodes.lua]: The whole lamp is flammable.
=> You loose normally not burnable stuff (the lamp part) if it catches fire.
Should this be taken as a disadvantage relative to the full mese lamp?

Comment (paramat) [inline mods/default/nodes.lua]: It could be non-flammable if preferred.

Comment (sofar) [inline mods/default/nodes.lua]: flammable is fine with me. The wooden part is the largest part of it, and players should take care with their expensive equipment and not set it on fire.

**Comment by paramat** [comment]
Updated texture subtly, the mese light section is 1 pixel taller which looks better to me.

**Comment by numberZero** [comment]
@paramat nice =)
3d inventory image (as for nodes) would be better IMO

**Comment by paramat** [comment]
New texture coming.

**Comment by paramat** [comment]
New texture, see first post, on request added more detail. The older texture is an alternative and is also shown in the first post.

**Comment by paramat** [comment]
Oops, new screenshots now actually added to first post.

**Comment by paramat** [comment]
Not sure about the 'mese post light' name. 'mese growlamp'? 'mese bollard light'? Dunno.

**Comment by numberZero** [comment]
@paramat It *is* a post light, so that’s fine for me. And that’s definitely not a `growlamp`, it would look differently.
But still, 3D inventory image looks nicer for me:
screenshot_20170203_022810
although it’s a bit too small. (The same for fences, gate’s 3D image is better than fence’s 2D). Also it better fits the modern MTG style IMO (we have 3D torches now after all!).

**Comment by paramat** [comment]
Oops i just realised it's easy to make it 3D in inventory, will do.

**Comment by paramat** [comment]
Done, thanks for pointing that out.
I'm fairly happy with this now so removed WIP.

**Comment by sofar** [comment]
:+1:

**Pull request merged**